README.md
iTagger
=======

Das Kurzfristige Ziel:
- Musikdateien halbautomatisch mit iTunes-Daten Taggen.<br>

Das Langfristige Ziel:
- Musik und Film/Serien-Dateien mit iTunes-Daten Taggen.
- Statistiken rund Um iTunes


Genutzte Bibliotheken:<br>
JSON.Net (http://json.codeplex.com)<br>
Taglib-sharp (https://github.com/sillsdev/taglib-sharp)<br>


taglib-sharp/NEWS
Version 2.1.1.0 (not yet official)
July 7, 2014

Support PNG files with compressed XMP data in the style of ImageMagick
Contributor: John Thomson
-------------------------------------------------------------------------------
Version 2.1.0.0
July 3, 2012

Fixes and Features in this release include:
  * Bump the version number because of an ABI break in 363f8ae
  * Initial support for Audible files (bgo#601543)

Contributors to this release:
  Alan McGovern, Alexander Kojevnikov, Guy Taylor

--------------------------------------------------------------------------------

Version 2.0.5.0
July 1, 2012

Fixes and Features in this release include:
  * Fix a freeze on some mp3 files where header was at the end of file
  * Properly implement partial matching for File.Find and File.RFind
  * [Mpeg4] Fix parsing of some apple tags (bgo#658920)
  * Add partial support for IPTC-IIM
  * Fix nested UDTA boxes
  * [Mpeg4] Attempt to handle corrupt tags a little better (bgo#676934)
  * Mark the file as corrupt instead of throwing exception (bgo#643477)
  * StartTag.ReadTag should not silence exceptions
  * Check for short popularimeter tags
  * On corrupt Id3v2 frame, skip just that frame (bgo#650674)
  * Use O(N) rather than O(N^2) deunsynchronization algorithm

Contributors to this release:
  Alan McGovern, Alexander Kojevnikov, Anton Drachev, Chow Loong Jin,
  Eamon Nerbonne, Eberhard Beilharz, Jakub 'Fiołek' Fijałkowski,
  Samuel D. Jack, Stephen Shaw, Tim Howard

--------------------------------------------------------------------------------

Version 2.0.4.0
April 13, 2011

Fixes and Features in this release include:
  * Support for Photo metadata
    + read/write for EXIF, XMP, PNG, JPEG, TIFF
	+ read support for many RAW formats
	+ see examples/ParsePhoto.cs
  * Support for Matroska metadata reading (eg WebM and .mkv)
  * Fix File.ReadBlock for slow connections (bgo#639817)

Contributors to this release:
  Alexander Kojevnikov, Colin Turner, Gabriel Burt, Jeroen Asselman,
  Julien Moutte, Marek Habersack, Mike Gemünde, Paul Lange,
  Ruben Vermeersch, Stephane Delcroix, Tim Howard

--------------------------------------------------------------------------------

Version 2.0.3.7
March 20, 2010

TagLib# bugs are now tracked at http://bugzilla.gnome.org.

Fixes and Features in this release include:
  * Solution/projects migrated to VS format
  * Fix ID3v2 unsyncing (bgo#593138)
  * Fix ID3v2.3 exetended header size calculation (bgo#604488)
  * Make TagLib.File IDisposable (closes any open streams)
  * Don't throw an exception if RVA2 data is partially incorrect
  * Fall back to nunit-console is nunit-console2 not found (bgo#593673)
  * Add build check for `al` tool (bgo#575303)

Contributors to this release:
  Alexander Kojevnikov, Andrés G. Aragoneses, Bertrand Lorentz,
  Eamon Nerbonne, Ruben Vermeersch

--------------------------------------------------------------------------------

Version 2.0.3.6
February 24, 2010

TagLib# bugs are now tracked at http://bugzilla.gnome.org.

Fixes and Features in this release include:
  * Reverse small API break introduced in 2.0.3.5

Contributors to this release:
  Gabriel Burt

--------------------------------------------------------------------------------

Version 2.0.3.5
February 23, 2010

TagLib# bugs are now tracked at http://bugzilla.gnome.org.

Fixes and Features in this release include:
  * Add BitsPerSample property (bgo#610585)
  * Support COMPILATION in ogg (bgo#586336)
  * Fix removing some/all tag-types (bgo#574653)

Contributors to this release:
  Alexander Kojevnikov, Helmut Wahrmann

--------------------------------------------------------------------------------

Version 2.0.3.4
January 26, 2010

TagLib# bugs are now tracked at http://bugzilla.gnome.org.

Get releases from http://download.banshee-project.org/taglib-sharp/

Fixes and Features in this release include:
  * Fix IndexOutOfRangeException (bgo#607376)
  * Fix issue with converting/parsing year (bgo#606283)
  * Support the 'alis' handler code/track type (bgo#584193)

--------------------------------------------------------------------------------

Version 2.0.3.3
September 30, 2009

TagLib# bugs are now tracked at http://bugzilla.gnome.org.

Get releases from http://download.banshee-project.org/taglib-sharp/

Fixes and Features in this release include:
  * Be more robust with tags with NULL bytes in the middle (bgo#558123)
  * Support AIFF file format (bgo#579261)
  * Support ADTS AAC files (bgo#580553)
  * Support for m2v files (bgo#574411)
  * Support for Sort* values in Ape tags (bgo#571421)
  * Add new properties: MusicBrainz, ArtistId, ReleaseId, ReleaseArtistId,
    TrackId, DiscId, ReleaseStatus, ReleaseType, ReleaseCountry, MusicIpId,
	and AmazonId (bgo#563170)
  * Add fallback options for album artist getter (bgo#574652)
  * Fixed bug with Ape.Tag's Pictures setter (bgo#575943)
  * Fix crashing when parsing some comment frames (bgo#582735)
  * Fixed CopyTo method (bgo#572382)
  * Fixed typo in Ogg DiscNumber setter that wrote TrackCount
    to the DISCTOTAL field (bgo#572382)
  * Fixed NRE when calling GetTag on FLAC file (bgo#572380)
  * Build improvements
  * API documentation fixes

Contributors to this release:
  Alexander Kojevnikov, Andy Beal, Bernd Niedergesaess, Bertrand Lorentz,
  Félix Velasco, Gabriel Burt, Gregory S. Chudov, Helmut Wahrmann, 
  John Millikin, Patrick Dehne

--------------------------------------------------------------------------------

Version 2.0.3.2
February 16, 2009

TagLib# bugs are now tracked at http://bugzilla.gnome.org.

Get releases from http://download.banshee-project.org/taglib-sharp/

Contributors to this release:
  Aaron Bockover, Andy Beal, Bernd Niedergesaess,
  Bertrand Lorentz, Gabriel Burt, John Millikin

Fixes in this release include:
  * Fix infinite loops when reading/writing video files
  * Fix parsing ID3v2 with Data Length Indicator set
  * Fix install/build issues with the docs
  * Fix casing of APE tags

taglib-sharp/src/AssemblyInfo.cs
//
// AssemblyInfo.cs.in: Contains flags to use for the assembly.
//
// Author:
//   Brian Nickel (brian.nickel@gmail.com)
//
// Copyright (C) 2006-2007 Brian Nickel
// 
// This library is free software; you can redistribute it and/or modify
// it  under the terms of the GNU Lesser General Public License version
// 2.1 as published by the Free Software Foundation.
//
// This library is distributed in the hope that it will be useful, but
// WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public
// License along with this library; if not, write to the Free Software
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
// USA
//

using System;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;

[assembly:AssemblyVersion("2.1.1.0")]
[assembly:AssemblyTitle ("TagLib#")]
[assembly:AssemblyDescription ("A library for reading and writing audio metatags.")]
[assembly:AssemblyCopyright ("Copyright (c) 2006-2007 Brian Nickel.  Copyright (c) 2009-2010 Other contributors")]
[assembly:AssemblyCompany ("")]
[assembly:AssemblyDelaySign(false)]
[assembly:AssemblyKeyFile("../taglib-sharp.snk")]
[assembly:CLSCompliant(false)]


taglib-sharp/src/TagLib/Mpeg4/BoxTypes.cs
//
// BoxTypes.cs: Contains common box names.
//
// Author:
//   Brian Nickel (brian.nickel@gmail.com)
//
// Copyright (C) 2006-2007 Brian Nickel
// 
// This library is free software; you can redistribute it and/or modify
// it  under the terms of the GNU Lesser General Public License version
// 2.1 as published by the Free Software Foundation.
//
// This library is distributed in the hope that it will be useful, but
// WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
// Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public
// License along with this library; if not, write to the Free Software
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
// USA
//

namespace TagLib.Mpeg4 {
	/// <summary>
	///    <see cref="BoxType" /> provides references to different box types
	///    used by the library.
	/// </summary>
	/// <remarks>
	///    <para>This class is used to severely reduce the number of times
	///    these types are created in <see cref="AppleTag" />, greatly
	///    improving the speed at which warm files are read.</para>
	///    <para>The reason it is marked as internal is because I'm not sure
	///    I like the way the fields are named, and it is really
	///    unneccessary for external uses. While the library may use
	///    <c>DataBoxes (BoxType.Gen, BoxType.Gnre);</c>, an external user
	///    could use <c>tag.DataBoxes ("gen", "gnre");</c> with the same
	///    result.</para>
	/// </remarks>
	internal static class BoxType
	{
		public static readonly ReadOnlyByteVector Aart = "aART";
		public static readonly ReadOnlyByteVector Alb  = AppleTag.FixId ("alb");
		public static readonly ReadOnlyByteVector Art  = AppleTag.FixId ("ART");
		public static readonly ReadOnlyByteVector Cmt  = AppleTag.FixId ("cmt");
		public static readonly ReadOnlyByteVector Cond = "cond";
		public static readonly ReadOnlyByteVector Covr = "covr";
		public static readonly ReadOnlyByteVector Co64 = "co64";
		public static readonly ReadOnlyByteVector Cpil = "cpil";
		public static readonly ReadOnlyByteVector Cprt = "cprt";
        public static readonly ReadOnlyByteVector Cnid = "cnid";
		public static readonly ReadOnlyByteVector Data = "data";
		public static readonly ReadOnlyByteVector Day  = AppleTag.FixId ("day");
		public static readonly ReadOnlyByteVector Disk = "disk";
		public static readonly ReadOnlyByteVector Esds = "esds";
		public static readonly ReadOnlyByteVector Ilst = "ilst";
		public static readonly ReadOnlyByteVector Free = "free";
		public static readonly ReadOnlyByteVector Gen  = AppleTag.FixId ("gen");
		public static readonly ReadOnlyByteVector Gnre = "gnre";
		public static readonly ReadOnlyByteVector Grp  = AppleTag.FixId("grp");
		public static readonly ReadOnlyByteVector Hdlr = "hdlr";
		public static readonly ReadOnlyByteVector Lyr  = AppleTag.FixId ("lyr");
		public static readonly ReadOnlyByteVector Mdat = "mdat";
		public static readonly ReadOnlyByteVector Mdia = "mdia";
		public static readonly ReadOnlyByteVector Meta = "meta";
		public static readonly ReadOnlyByteVector Mean = "mean";
		public static readonly ReadOnlyByteVector Minf = "minf";
		public static readonly ReadOnlyByteVector Moov = "moov";
		public static readonly ReadOnlyByteVector Mvhd = "mvhd";
		public static readonly ReadOnlyByteVector Nam  = AppleTag.FixId ("nam");
		public static readonly ReadOnlyByteVector Name = "name";
        public static readonly ReadOnlyByteVector Rtng = "rtng";
		public static readonly ReadOnlyByteVector Skip = "skip";
		public static readonly ReadOnlyByteVector Soaa = "soaa"; // Album Artist Sort
		public static readonly ReadOnlyByteVector Soar = "soar"; // Performer Sort
		public static readonly ReadOnlyByteVector Soco = "soco"; // Composer Sort
		public static readonly ReadOnlyByteVector Sonm = "sonm"; // Track Title Sort
		public static readonly ReadOnlyByteVector Soal = "soal"; // Album Title Sort
		public static readonly ReadOnlyByteVector Stbl = "stbl";
		public static readonly ReadOnlyByteVector Stco = "stco";
		public static readonly ReadOnlyByteVector Stsd = "stsd";
		public static readonly ReadOnlyByteVector Text = "text";
		public static readonly ReadOnlyByteVector Tmpo = "tmpo";
		public static readonly ReadOnlyByteVector Trak = "trak";
		public static readonly ReadOnlyByteVector Trkn = "trkn";
		public static readonly ReadOnlyByteVector Udta = "udta";
		public static readonly ReadOnlyByteVector Url = AppleTag.FixId ("url");
		public static readonly ReadOnlyByteVector Uuid = "uuid";
		public static readonly ReadOnlyByteVector Wrt  = AppleTag.FixId ("wrt");
		public static readonly ReadOnlyByteVector DASH = "----";

		// Handler types.
		public static readonly ReadOnlyByteVector Soun = "soun";
		public static readonly ReadOnlyByteVector Vide = "vide";

		// Another handler type, found in wild in audio file ripped using iTunes
		public static readonly ReadOnlyByteVector Alis = "alis";
	}
}
